torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -356,6 +384,16 @@ TORCH_LIBRARY(_c10d_functional, m) {
           c10::DispatchKey::CompositeExplicitAutograd, ::broadcast_),
       {at::Tag::pt2_compliant_tag});
 
+  m.def(
+      "scatter(Tensor input, Tensor[] scatter_list, int src, str group_name) -> Tensor",
```

Comment:
The name `input` looks very confusing here tbh, since it's not used as input in either cases.

- Can we do `SymInt[] size` + `Device device` in the out-of-place variant here?
- For the in-place variant, can we change `input` to `output`?

Reply:
I think we should stick to the https://pytorch.org/docs/stable/distributed.html#torch.distributed.scatter interface as much as possible, agreed the naming input is confusing, will update

Reply:
renamed input to match distributed api `tensor`

Reply:
Thanks! `scatter`'s a bit of a weirdo compared to the others. `tensor` sounds good to me. Better keeping it vague than explicitly misleading :)